controllers/secret.go

```diff
@@ -96,18 +96,47 @@ func (r *OperatorPipelineReconciler) reconcileGitHubAPISecret(meta metav1.Object
 
 // reconcilePyxisAPISecret will ensure that the Pyxis API Secret is present and up to date.
 func (r *OperatorPipelineReconciler) reconcilePyxisAPISecret(meta metav1.ObjectMeta) error {
+	pipelineName := types.NamespacedName{
+		Namespace: meta.Namespace,
+		Name:      meta.Name,
+	}
+	pipeline := &certv1alpha1.OperatorPipeline{}
+	err := r.Client.Get(context.TODO(), pipelineName, pipeline)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Info("pipeline resource not found. Ignoring since object must be deleted")
+			return nil
+		}
+		log.Error(err, "unable to retrieve pipeline resource in namespace "+meta.Namespace)
+		return err
+	}
+	var secretName = PYXIS_API_SECRET
+	if pipeline.Spec.PyxisApiSecretName != "" {
+		secretName = pipeline.Spec.PyxisApiSecretName
+	}
 	key := types.NamespacedName{
 		Namespace: meta.Namespace,
-		Name:      PYXIS_API_SECRET,
+		Name:      secretName,
 	}
-
 	secret := newSecret(key)
-	if IsObjectFound(r.Client, key, secret) {
-		log.Info("existing pyxis api secret found")
-		return nil // Existing Secret found, do nothing...
+	if !IsObjectFound(r.Client, key, secret) {
+		err := errors.New(fmt.Sprintf("an existing secret named %s was not found in namespace %s", secretName, meta.Namespace))
+		log.Error(err, fmt.Sprintf("unable to reconcile %s in namespace %s", secretName, meta.Namespace))
+		return err
 	}
-
-	return errors.New("pyxis api secret not found")
+	log.Info(fmt.Sprintf("existing %s secret found in namespace %s", secretName, meta.Namespace))
+	kubeConfigSecret := &corev1.Secret{}
+	err = r.Client.Get(context.TODO(), key, kubeConfigSecret)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to get the %s secret", secretName))
+		return err
+	}
+	if kubeConfigSecret.Data[KUBECONFIG_SECRET] == nil {
```

Comment:
This references the `kubeconfig` value. the value for this key should be
`PYXIS_API_KEY     = "pyxis_api_key"`

Also. checking in a map this way can throw a panic if the key isn't present.
```
if value, ok := kubeConfigSecret.Data[KUBECONFIG_SECRET]; ok{
//then evaluate data in value
}
```

Reply:
I think if the requested key doesn’t exist, we get the value type’s zero value..

Reply:
Missed that was the same as the secret name in the other two cases.. i think.